app/components/post_list/post/body/content/permalink_preview/index.tsx

```diff
@@ -8,14 +8,15 @@ import {switchMap, distinctUntilChanged} from 'rxjs/operators';
 import {fetchUsersByIds} from '@actions/remote/user';
 import {withServerUrl} from '@context/server';
 import {observePost} from '@queries/servers/post';
+import {observeCanDownloadFiles, observeEnableSecureFilePreview} from '@queries/servers/security';
 import {observeConfigBooleanValue} from '@queries/servers/system';
 import {observeUserOrFetch, observeTeammateNameDisplay, observeCurrentUser} from '@queries/servers/user';
 
 import PermalinkPreview from './permalink_preview';
 
 import type {WithDatabaseArgs} from '@typings/database/database';
 
-const enhance = withObservables(['embedData', 'serverUrl'], ({database, embedData, serverUrl}: WithDatabaseArgs & {embedData: PermalinkEmbedData; serverUrl?: string}) => {
+const enhance = withObservables(['embedData', 'serverUrl'], ({database, embedData, serverUrl, parentLocation, parentPostId}: WithDatabaseArgs & {embedData: PermalinkEmbedData; serverUrl?: string; parentLocation?: string; parentPostId?: string}) => {
```

Comment:
Two things here. One was "wrong" from before. In order to keep things cleaner, let's create a "OwnProps" type. That will be...

```
type OwnProps = WithDatabaseArgs & {
    embedData: PermalinkEmbedData;
    serverUrl?: string;
    parentLocation?: string;
    parentPostId?: string;
};
```

The other thing is that there is no need to add `parentLocation` and `parentPostId` here. You are not using them to calculate any observables. It should just be added to the other component props, and it should be passed directly.

Reply:
Ah yes, nice catch.